### ISSUE TITLE
NPE on accessing isSeparateMedia

### ISSUE THREAD

**Issue by henning-gerhardt** [kitodo/kitodo-production#5065]
While working on newspaper year processes a NPE is sometimes caused on using method isSeparateMedia() of class StructurePanel. This is happening in most cases on closing meta data editor and looks like

```
Request URI: /kitodo/pages/metadataEditor.jsf
Ajax request: Yes
Status code: 500
Exception type: class java.lang.NullPointerException
Exception message:
Exception UUID:
Stack trace:
java.lang.NullPointerException
               at org.kitodo.production.forms.dataeditor.StructurePanel.isSeparateMedia(StructurePanel.java:1379)
               at jdk.internal.reflect.GeneratedMethodAccessor608.invoke(Unknown Source)
               at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
               at java.base/java.lang.reflect.Method.invoke(Method.java:566)
               at javax.el.BeanELResolver.getValue(BeanELResolver.java:88)
               at javax.el.CompositeELResolver.getValue(CompositeELResolver.java:62)
               at org.apache.myfaces.el.unified.resolver.FacesCompositeELResolver.getValue(FacesCompositeELResolver.java:179)
               at org.apache.el.parser.AstValue.getValue(AstValue.java:169)
               at org.apache.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:190)
               at org.jboss.weld.el.WeldValueExpression.getValue(WeldValueExpression.java:50)
               at org.apache.myfaces.view.facelets.el.ContextAwareTagValueExpression.getValue(ContextAwareTagValueExpression.java:93)
               at javax.faces.component._DeltaStateHelper.eval(_DeltaStateHelper.java:372)
               at javax.faces.component.UIComponentBase.isRendered(UIComponentBase.java:1212)
...
```

The stack trace appears even inside the catalina.out log file of Tomcat:

```
[crit] FullAjaxExceptionHandler: An exception occurred during processing JSF ajax request. Error page '/pages/error.jsf' will be shown.
[crit] java.lang.NullPointerException
[crit]     at org.kitodo.production.forms.dataeditor.StructurePanel.isSeparateMedia(StructurePanel.java:1379)
[crit]     at jdk.internal.reflect.GeneratedMethodAccessor608.invoke(Unknown Source)
[crit]     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[crit]     at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[crit]     at javax.el.BeanELResolver.getValue(BeanELResolver.java:88)
[crit]     at javax.el.CompositeELResolver.getValue(CompositeELResolver.java:62)
[crit]     at org.apache.myfaces.el.unified.resolver.FacesCompositeELResolver.getValue(FacesCompositeELResolver.java:179)
[crit]     at org.apache.el.parser.AstValue.getValue(AstValue.java:169)
[crit]     at org.apache.el.ValueExpressionImpl.getValue(ValueExpressionImpl.java:190)
[crit]     at org.jboss.weld.el.WeldValueExpression.getValue(WeldValueExpression.java:50)
[crit]     at org.apache.myfaces.view.facelets.el.ContextAwareTagValueExpression.getValue(ContextAwareTagValueExpression.java:93)
[crit]     at javax.faces.component._DeltaStateHelper.eval(_DeltaStateHelper.java:372)
[crit]     at javax.faces.component.UIComponentBase.isRendered(UIComponentBase.java:1212)
...
```

As this NPE did not occur every time and the messages did not appear every time (according to the logs this error happened more times than visible) I can not provide more information.

**Comment by henning-gerhardt** [comment]
The not modified code was already checking if a process has a template assigned but there was not a check if the dataeditorform class itself has a process assigned as a process is not mandatory for the dataeditorform class. 

I looked into the database too but all processes has a template assigned so the NPE could only occur if the dataeditorform class has not a process assigned.

**Comment by henning-gerhardt** [comment]
Closing as fixed by #5066 .
